api/bucketsd/service.go

```diff
@@ -3218,32 +3218,37 @@ func (s *Service) Archive(ctx context.Context, req *pb.ArchiveRequest) (*pb.Arch
 			}
 		} else { // Case 2.
 			res, err := s.PowergateClient.Data.CidInfo(ctxPow, oldCid.String())
-			if err != nil {
+			isKnownToPowergate := err == nil || status.Convert(err).Code() != codes.NotFound
+			if err != nil && isKnownToPowergate {
 				return nil, fmt.Errorf("looking up old storage config: %v", err)
 			}
 
-			if cmp.Equal(&storageConfig, res.CidInfo.LatestPushedStorageConfig) {
+			if isKnownToPowergate && cmp.Equal(&storageConfig, res.CidInfo.LatestPushedStorageConfig) {
 				// Old storage config is the same as the new so use replace.
 				res, err := s.PowergateClient.Data.ReplaceData(ctxPow, oldCid.String(), p.Cid().String())
 				if err != nil {
 					return nil, fmt.Errorf("replacing cid: %v", err)
 				}
 				jid = res.JobId
 			} else {
-				// New storage config, so remove and push.
-				_, err = s.PowergateClient.StorageConfig.Apply(
-					ctxPow,
-					oldCid.String(),
-					pow.WithStorageConfig(&userPb.StorageConfig{}),
-					pow.WithOverride(true),
-				)
-				if err != nil {
-					return nil, fmt.Errorf("pushing config to disable hot and cold storage: %v", err)
-				}
-				_, err = s.PowergateClient.StorageConfig.Remove(ctxPow, oldCid.String())
-				if err != nil {
-					return nil, fmt.Errorf("removing old cid storage: %v", err)
+				// New storage config.
+				if isKnownToPowergate {
+					// Remove previous storage config.
+					_, err = s.PowergateClient.StorageConfig.Apply(
+						ctxPow,
+						oldCid.String(),
+						pow.WithStorageConfig(&userPb.StorageConfig{}),
+						pow.WithOverride(true),
+					)
+					if err != nil {
+						return nil, fmt.Errorf("pushing config to disable hot and cold storage: %v", err)
+					}
+					_, err = s.PowergateClient.StorageConfig.Remove(ctxPow, oldCid.String())
+					if err != nil {
+						return nil, fmt.Errorf("removing old cid storage: %v", err)
+					}
```

Comment:
In this case, we are applying a new storage config AND the cid is known to powergate, so we can do the remove step.

Reply:
Then below, outside of this `if`, we always apply the new storage config.